sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/RedundantConditionalAroundAssignment.cs

```diff
@@ -48,7 +49,8 @@ protected sealed override void Initialize(SonarAnalysisContext context)
                 {
                     var ifStatement = (IfStatementSyntax)c.Node;
                     if (ifStatement.Else != null ||
-                        ifStatement.Parent is ElseClauseSyntax)
+                        ifStatement.Parent is ElseClauseSyntax ||
+                        ifStatement.FirstAncestorOrSelf<PropertyDeclarationSyntax>() is PropertyDeclarationSyntax)

```

Comment:
... here

Reply:
Actually I don't know if it is wrong or not but this looks really weird...

Reply:
That should work fine. I'll double-check on peach just in case.